### PR TITLE
Eable loading of an app domain manager from a stream

### PR DESCRIPTION
**Pull request by moozzyk**
Needed to be able to redirect to a different runtime for servicing purposes.

### CODE REVIEW

**Comment by moozzyk** [comment]
/cc @davidfowl @lodejard 

@jkotas - I would appreciate if you could also take a look at this

**Comment by moozzyk** [comment]
@blowdart - this will need security review

File: src/dnx.clr/HostAssemblyStore.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
+#include "stdafx.h"
+#include "HostAssemblyStore.h"
+#include "FileStream.h"
+#include "ComObject.h"
+#include <string>
+
+const wchar_t* AppDomainManagerTypeName = L"dnx.clr.managed, Version=1.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60, ProcessorArchitecture=MSIL";
```
Comment (moozzyk) [inline src/dnx.clr/HostAssemblyStore.cpp]: `AppDomainManagerTypeName` :arrow_right: `AppDomainManagerAssemblyName`

**Comment by moozzyk** [comment]
:watch:

**Comment by jkotas** [comment]
The implementation looks fine to me.

You should watch out for unexpected issues ... HostAssemblyManager is a corner case feature that was build and tested for very specific scenario. It is used it a new mode here that it was not used before.

**Comment by moozzyk** [comment]
Thanks @jkotas!

**Pull request closed without merging**